### PR TITLE
Track last successful record

### PR DESCRIPTION
**Pull request by CrossfireCurt**
This is a branch of the "pump_client" branch, so please merge https://github.com/commercehub-oss/watershed/pull/60 first.

The last successfully processed record will now be part of the Job. In the case of an error, it will be nice to see the original record so an operator could use that in the next query issued if they wanted to not rerun everything up to the point of error.

- Wraps Drill record and Kinesis record in a new object: PumpRecord
- Pump now strictly works off of PumpRecords, generating PumpRecordResults
- When the subscriber gets notice that a record is successful, the Job is updated with Drill record as the lastSuccessfulRow

### CODE REVIEW

File: pump/src/main/java/com/commercehub/watershed/pump/processing/PumpSubscriber.java [hunk 1]
```diff
@@ -97,10 +101,11 @@ public void onError(Throwable e) {
     }
 
     @Override
-    public void onNext(UserRecordResult userRecordResult) {
+    public void onNext(PumpRecordResult pumpRecordResult) {
         log.trace("Got a Kinesis result.");
-        if (userRecordResult.isSuccessful()) {
+        if (pumpRecordResult.getUserRecordResult().isSuccessful()) {
             successCount.incrementAndGet();
+            lastSuccessfulRow = pumpRecordResult.getResultRow();
```
Comment (aheiss1) [inline pump/src/main/java/com/commercehub/watershed/pump/processing/PumpSubscriber.java]: This is a little confusing, a naming issue.  The ResultRow gathered from a PumpRecordResult is not actually the overall result.  It's a corresponding/underlying Drill ResultRow that relates in the PumpRecordResult.  Suggestions include renaming ResultRow to a DrillResultRow, and/or renaming the member variable of PumpRecordResult to be something like 'drillRow'.  In that case, what we see here would change to pumpRecordResult.getDrillRow() which is either a ResultRow or DrillResultRow (depending on if we choose to rename ResultRow).  I'd tend to want to rename to DrillResultRow, as for whatever reasons ResultRow within an app that uses java.sql caused me to take a double take at what namespace that object was in (ResultRow is not actually a java.sql object, but thought it was).

File: pump/src/main/java/com/commercehub/watershed/pump/model/PumpRecordResult.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.commercehub.watershed.pump.model;
+
+import com.amazonaws.services.kinesis.producer.UserRecordResult;
+
+/**
+ * Holds results for the KinesisRecord (post-emission) and the row that was pulled from Drill.
+ */
+public class PumpRecordResult {
+    private UserRecordResult userRecordResult;
+    private ResultRow resultRow;
```
Comment (aheiss1) [inline pump/src/main/java/com/commercehub/watershed/pump/model/PumpRecordResult.java]: Naming is inconsistent with member variable of PumpRecord drillRow.  Was this on purpose?

**Comment by aheiss1** [comment]
Please remerge with master branch.

**Comment by CrossfireCurt** [comment]
@aheiss1 all set!

**Pull request merged**